exp/api/v1beta2/rosamachinepool_types.go

```diff
@@ -79,7 +80,7 @@ type RosaMachinePoolSpec struct {
 	// Autoscaling specifies auto scaling behaviour for this MachinePool.
 	// required if Replicas is not configured
 	// +optional
-	Autoscaling *RosaMachinePoolAutoScaling `json:"autoscaling,omitempty"`
+	Autoscaling *rosacontrolplanev1.RosaMachinePoolAutoScaling `json:"autoscaling,omitempty"`
```

Comment:
I believe the one created above to avoid loop dependency, so keep defining the  RosaMachinePoolAutoScaling within the RosaMachinePool API is better to min the ref . RosaMachinePoolAutoScaling is a simple struct to duplicate

Reply:
This is not adding new dependency. We need `rosacontrolplanev1.ExternalAuthProvider` so we can also import RosaMachinePoolAutoScaling.